diff -ur mgbare/MG5_aMC_v2_1_1/madgraph/interface/madgraph_interface.py MG5_aMC_v2_1_1/madgraph/interface/madgraph_interface.py
--- mgbare/MG5_aMC_v2_1_1/madgraph/interface/madgraph_interface.py      2014-04-01 09:44:13.000000001 +0200
+++ MG5_aMC_v2_1_1/madgraph/interface/madgraph_interface.py     2014-05-29 20:12:08.000000001 +0200
@@ -4290,12 +4290,14 @@
                 ld_path = misc.Popen([self.options['lhapdf'], '--libdir'],
                                      stdout=subprocess.PIPE).communicate()[0]
                 ld_path = ld_path.replace('\n','')
-                if 'LD_LIBRARY_PATH' not in os.environ:
-                    os.environ['LD_LIBRARY_PATH'] = ld_path
-                elif not os.environ['LD_LIBRARY_PATH']:
-                    os.environ['LD_LIBRARY_PATH'] = ld_path
-                elif 1:#ld_path not in os.environ['LD_LIBRARY_PATH']:
-                    os.environ['LD_LIBRARY_PATH'] += ';%s' % ld_path
+                lhaincludes = misc.Popen([self.options['lhapdf'], '--incdir'],
+                                     stdout=subprocess.PIPE).communicate()[0]
+                lhaincludes = lhaincludes.replace('\n','')                    
+                path = os.path.join(MG5DIR, 'SysCalc', 'src', 'Makefile')
+                text = open(path).read()
+                text = text.replace('INCLUDES =  -I../include','INCLUDES =  -I../include -I%s' % (lhaincludes))
+                text = text.replace('LIBS = -lLHAPDF','LIBS = %s/libLHAPDF.a -lgfortran' % (ld_path))
+                open(path, 'w').writelines(text)                    
             else:
                 raise self.InvalidCmd('lhapdf is required to compile/use SysCalc')
 
diff -ur mgbare/MG5_aMC_v2_1_1/madgraph/various/cluster.py MG5_aMC_v2_1_1/madgraph/various/cluster.py
--- mgbare/MG5_aMC_v2_1_1/madgraph/various/cluster.py   2014-04-01 09:44:13.000000001 +0200
+++ MG5_aMC_v2_1_1/madgraph/various/cluster.py  2014-05-29 15:29:28.000000001 +0200
@@ -206,8 +206,8 @@
         while 1: 
             nb_iter += 1
             idle, run, finish, fail = self.control(me_dir)
-            if fail:
-                raise ClusterManagmentError('Some Jobs are in a Hold/... state. Please try to investigate or contact the IT team')
+#            if fail:
+#                raise ClusterManagmentError('Some Jobs are in a Hold/... state. Please try to investigate or contact the IT team')
             if idle + run == 0:
                 #time.sleep(20) #security to ensure that the file are really written on the disk
                 logger.info('All jobs finished')
@@ -1273,6 +1273,9 @@
         if log is None:
             log = '/dev/null'
         
+        text += 'OLDPWD=$PWD;'
+        text += 'if [ -n $CMSSW_BASE ]; then cd $CMSSW_BASE; eval `scramv1 runtime -sh`; cd $OLDPWD; fi;'
+        
         text += prog
         if argument:
             text += ' ' + ' '.join(argument)

=== modified file 'Template/NLO/Source/ranmar.f'
--- MG5_aMC_v2_1_1/Template/NLO/Source/ranmar.f	2012-08-30 14:59:00 +0000
+++ MG5_aMC_v2_1_1/Template/NLO/Source/ranmar.f	2014-06-12 08:28:28 +0000
@@ -50,7 +50,12 @@
       if (init .eq. 1) then
          init = 0
          call get_offset(ioffset)
-         if (iseed .eq. 0) call get_base(iseed)
+cRF : always read the seed from the randinit file (this file is updated
+c by amcatnlo_run_interface every time a run starts). This makes sure
+c that the code does not need any remcompilation when only the seed is
+c changed (useful for NLO gridpack mode).
+c$$$         if (iseed .eq. 0) call get_base(iseed)
+         call get_base(iseed)
 c
 c     TJS 3/13/2008
 c     Modified to allow for more sequences 

=== modified file 'Template/NLO/SubProcesses/ajob_template'
--- MG5_aMC_v2_1_1/Template/NLO/SubProcesses/ajob_template	2014-03-13 18:02:02 +0000
+++ MG5_aMC_v2_1_1/Template/NLO/SubProcesses/ajob_template	2014-06-12 08:34:09 +0000
@@ -62,12 +62,13 @@
             fi
         fi
     fi
-    if  [[ ! -e randinit ]] ; then
-        if [[ -e ../randinit ]] ; then
-            cp ../randinit .
-        else
-            cp ../../randinit .
-        fi
+    if  [[ -e randinit ]] ; then
+	rm -f randinit
+    fi
+    if [[ -e ../randinit ]] ; then
+        ln -sf ../randinit .
+    else
+        ln -sf ../../randinit .
     fi
     if  [[ ! -e symfact.dat ]] ; then
     	if [[  -e ../symfact.dat ]] ; then
