diff -ruN ./src/randomgenerator.cpp ../UPDATED/src/randomgenerator.cpp
--- ./src/randomgenerator.cpp	2019-11-20 14:43:12.000000001 +0100
+++ ../UPDATED/src/randomgenerator.cpp	2023-05-22 08:34:56.000000001 +0200
@@ -123,7 +123,7 @@
    const unsigned int kMatrixA =         0x9908b0df;
 
    if (_count624 >= kN) {
-      register int i;
+      int i;
 
       for (i=0; i < kN-kM; i++) {
          y = (_Mt[i] & kUpperMask) | (_Mt[i+1] & kLowerMask);
