--- a/madgraph/various/banner.py	2023-03-28 23:08:20.692089612 +0200
+++ b/madgraph/various/banner.py	2023-03-28 23:08:15.083542354 +0200
@@ -4286,8 +4286,8 @@
 
         # for lepton-lepton collisions, ignore 'pdlabel' and 'lhaid'
         if abs(self['lpp1'])!=1 or abs(self['lpp2'])!=1:
-            if self['lpp1'] == 1 or self['lpp2']==1:
-                raise InvalidRunCard('Process like Deep Inelastic scattering not supported at NLO accuracy.')
+       #     if self['lpp1'] == 1 or self['lpp2']==1:
+       #         raise InvalidRunCard('Process like Deep Inelastic scattering not supported at NLO accuracy.')
             
             if self['pdlabel']!='nn23nlo' or self['reweight_pdf']:
                 self['pdlabel']='nn23nlo'
