--- a/madgraph/interface/loop_interface.py	2019-07-06 03:19:30.000000002 +0200
+++ b/madgraph/interface/loop_interface.py	2019-07-06 03:22:46.000000002 +0200
@@ -501,7 +501,7 @@
         opt = self.options
                 
         # Check if first time:
-        if not force and ((opt['ninja'] is None) or (os.path.isfile(pjoin(MG5DIR, opt['ninja'],'libninja.a')))): 
+        if not force and ((opt['ninja'] is None) or (os.path.isfile(pjoin(os.path.abspath(opt['ninja']),'libninja.a')))):    
             return
 
         # do not trigger the question for tests
