--- a/madgraph/various/systematics.py	2017-05-01 21:02:10.402600552 +0300
+++ b/madgraph/various/systematics.py	2017-05-01 21:01:44.759601146 +0300
@@ -228,7 +228,7 @@
         else:
             lowest_id = self.get_id()        
 
-        ids = [lowest_id+i for i in range(len(self.args)-1)]
+        ids = [lowest_id+i for i in range(len(self.args))]
         all_cross = [0 for i in range(len(self.args))]
         
         self.input.parsing = False
@@ -261,7 +261,7 @@
                 print event
                 raise Exception
             
-            wgt = [event.wgt*wgts[i]/wgts[0] for i in range(1,len(wgts))]
+            wgt = [event.wgt*wgts[i]/wgts[0] for i in range(0,len(wgts))]
             all_cross = [(all_cross[j] + event.wgt*wgts[j]/wgts[0]) for j in range(len(wgts))]
             
             rwgt_data = event.parse_reweight()
@@ -416,7 +416,7 @@
         text = ''
         
         default = self.args[0]
-        for arg in self.args[1:]:
+        for arg in self.args:
             mur, muf, alps, dyn, pdf = arg[:5]
             if pdf == self.orig_pdf and alps ==1 and (mur!=1 or muf!=1 or dyn!=-1):
                 if not in_scale:
         
         
