--- a/madgraph/interface/common_run_interface.py   2016-11-04 10:49:37.000000000 +0100
+++ b/madgraph/interface/common_run_interface.py   2016-11-23 09:04:08.721495000 +0100
@@ -1534,7 +1535,10 @@

         # always pass to a path + get the event size
         result_file= sys.stdout
-        if not os.path.sep in args[0]:
+
+        if os.path.isfile(args[0]) :
+            print 'have file name as input'
+        elif not os.path.sep in args[0]:
             path = [pjoin(self.me_dir, 'Events', args[0], 'unweighted_events.lhe.gz'),
                     pjoin(self.me_dir, 'Events', args[0], 'unweighted_events.lhe'),
                     pjoin(self.me_dir, 'Events', args[0], 'events.lhe.gz'),
