From ddfc18a9d88b70282ad7322450e8ff8578062874 Mon Sep 17 00:00:00 2001
From: Josh Bendavid <Josh.Bendavid@cern.ch>
Date: Tue, 28 Jun 2016 23:03:49 +0200
Subject: [PATCH 8/8] allow for reweighting with up to 25 PDF sets (for NLO process generation)

---
 Template/LO/Source/make_opts            |    8 ++++----

diff --git a/Template/LO/Source/make_opts b/Template/LO/Source/make_opts
index 92a3fe4..a2c75f7 100644
--- a/Template/LO/Source/make_opts
+++ b/Template/LO/Source/make_opts
@@ -1,9 +1,9 @@
 DEFAULT_F2PY_COMPILER=f2py
 DEFAULT_F_COMPILER=gfortran
-MACFLAG=-mmacosx-version-min=10.7
-DEFAULT_CPP_COMPILER=clang
-STDLIB=-lc++
-STDLIB_FLAG=-stdlib=libc++
+MACFLAG=
+DEFAULT_CPP_COMPILER=g++
+STDLIB=-lstdc++
+STDLIB_FLAG=
 #end_of_make_opts_variables
 # Rest of the makefile
