*** a/madgraph/iolibs/template_files/madevent_makefile_source	2016-11-04 10:49:37.000000000 +0100
--- b/madgraph/iolibs/template_files/madevent_makefile_source	2016-11-18 13:14:15.908317000 +0100
***************
*** 57,58 ****
! $(BINDIR)combine_events: $(COMBINE) $(LIBDIR)libmodel.$(libext) $(LIBDIR)libpdf.$(libext) run_card.inc
! 			 $(FC) $(LDFLAGS) -o $@ $(COMBINE) -L$(LIBDIR) -lmodel -lpdf $(llhapdf) $(LDFLAGS)
--- 57,58 ----
! $(BINDIR)combine_events: $(COMBINE) $(LIBDIR)libmodel.$(libext) $(LIBDIR)libpdf.$(libext) run_card.inc $(LIBDIR)libbias.$(libext)
! 			 $(FC) $(LDFLAGS) -o $@ $(COMBINE) -L$(LIBDIR) -lmodel -lpdf $(llhapdf) $(LDFLAGS) -lbias

--- a/Template/LO/Cards/run_card.dat	2016-11-04 10:49:37.000000001 +0100
+++ b/Template/LO/Cards/run_card.dat	2016-11-22 09:29:33.000000001 +0100
@@ -276,6 +276,6 @@
 %(sys_alpsfact)s = sys_alpsfact  # \alpha_s emission scale factors
 %(sys_matchscale)s = sys_matchscale # variation of merging scale
 # PDF sets and number of members (0 or none for all members).
-%(sys_pdf)s = sys_pdf # separate by && if more than one set.
+%(sys_pdf)s = sys_pdf # separate by two spaces if more than one set.
 # MSTW2008nlo68cl.LHgrid 1  = sys_pdf
 
