commit 554d9f95b18b0cf265cafe0b28e3a0eb7898c53b
Author: Alberto Sanchez Hernandez <alberto.sanchez@mail.com>
Date:   Thu Sep 28 02:04:02 2017 +0200

    fix lhapdf and boost issues

diff --git a/Template/LO/Source/.make_opts b/Template/LO/Source/.make_opts
index 30da110..430c038 100644
--- a/Template/LO/Source/.make_opts
+++ b/Template/LO/Source/.make_opts
@@ -96,6 +96,7 @@ endif
 
 ifneq ($(lhapdf),)
   CXXFLAGS += $(shell $(lhapdf) --cppflags)
+  CXXFLAGS += -I$(BOOSTINCLUDES)
   alfas_functions=alfas_functions_lhapdf
   llhapdf+= $(shell $(lhapdf) --cflags --libs) -lLHAPDF
 # check if we need to activate c++11 (for lhapdf6.2)
diff --git a/Template/LO/Source/PDF/pdf_lhapdf6.cc b/Template/LO/Source/PDF/pdf_lhapdf6.cc
index 35b34d5..77daf67 100644
--- a/Template/LO/Source/PDF/pdf_lhapdf6.cc
+++ b/Template/LO/Source/PDF/pdf_lhapdf6.cc
@@ -12,6 +12,11 @@
 #include "LHAPDF/Version.h"
 #include "LHAPDF/LHAGlue.h"
 
+#include <boost/shared_ptr.hpp>
+#include <boost/foreach.hpp>
+#include <boost/algorithm/string/case_conv.hpp>
+#include <cstring>
+
 using namespace std;
 
 
diff --git a/Template/NLO/Source/PDF/pdf_lhapdf6.cc b/Template/NLO/Source/PDF/pdf_lhapdf6.cc
index 35b34d5..77daf67 100644
--- a/Template/NLO/Source/PDF/pdf_lhapdf6.cc
+++ b/Template/NLO/Source/PDF/pdf_lhapdf6.cc
@@ -12,6 +12,11 @@
 #include "LHAPDF/Version.h"
 #include "LHAPDF/LHAGlue.h"
 
+#include <boost/shared_ptr.hpp>
+#include <boost/foreach.hpp>
+#include <boost/algorithm/string/case_conv.hpp>
+#include <cstring>
+
 using namespace std;
 
 
diff --git a/Template/NLO/Source/make_opts.inc b/Template/NLO/Source/make_opts.inc
index f41a7eb..4b03a8b 100644
--- a/Template/NLO/Source/make_opts.inc
+++ b/Template/NLO/Source/make_opts.inc
@@ -86,6 +86,7 @@ endif
 
 ifneq ($(lhapdf),)
   CXXFLAGS += $(shell $(lhapdf) --cppflags)
+  CXXFLAGS += -I$(BOOSTINCLUDES)
   alfas_functions=alfas_functions_lhapdf
   llhapdf = $(shell $(lhapdf) --libs)
   reweight_xsec_events_pdf_dummy=
diff --git a/madgraph/iolibs/template_files/pdf_wrap_lhapdf.f b/madgraph/iolibs/template_files/pdf_wrap_lhapdf.f
index b06677d..0a53fd6 100644
--- a/madgraph/iolibs/template_files/pdf_wrap_lhapdf.f
+++ b/madgraph/iolibs/template_files/pdf_wrap_lhapdf.f
@@ -20,7 +20,7 @@ c-------------------
 
 c     initialize the pdf set
       call FindPDFPath(LHAPath)
-      CALL SetPDFPath(LHAPath)
+c      CALL SetPDFPath(LHAPath)
       value(1)=lhaid
       parm(1)='DEFAULT'
       call pdfset(parm,value)
