--- a/MadSpin/interface_madspin.py 
+++ b/MadSpin/interface_madspin.py
@@ -1021,7 +1021,7 @@
                     
                     run_card["iseed"] = self.seed
                     run_card['gridpack'] = True
-                    run_card['systematics_program'] = False
+                    run_card['use_syst'] = False
                     run_card.write(pjoin(decay_dir, "Cards", "run_card.dat"))
                     param_card = self.banner['slha']
                     open(pjoin(decay_dir, "Cards", "param_card.dat"),"w").write(param_card)
@@ -1462,8 +1462,8 @@
             mymod = getattr(mymod, 'matrix2py')  
             with misc.chdir(pjoin(self.path_me, 'madspin_me', 'SubProcesses', pdir)):
                 with misc.stdchannel_redirected(sys.stdout, os.devnull):
-                    mymod.initialise(pjoin(self.path_me, 'Cards','param_card.dat'))
-            self.all_f2py[pdir] = mymod.get_me  
+                    mymod.initialisemodel(pjoin(self.path_me, 'Cards','param_card.dat'))
+            self.all_f2py[pdir] = mymod.get_value
             return self.calculate_matrix_element(event)
         
         
