diff --git a/madgraph/various/cluster.py b/madgraph/various/cluster.py
--- a/madgraph/various/cluster.py
+++ b/madgraph/various/cluster.py
@@ -1216,1 +1216,1 @@
-    maximum_submited_jobs = 2500
+    maximum_submited_jobs = 1000
@@ -1244,2 +1244,3 @@
+        text += "source /cvmfs/cms.cern.ch/cmsset_default.sh;eval `scram runtime -sh`;"
         if not os.path.isabs(prog):
             text += "./%s" % prog
@@ -1281,6 +1282,9 @@
-        for line in status.stdout:
+        for line in (status.stdout + status.stderr):
             line = line.strip()
             if 'cannot connect to server' in line or 'cannot read reply' in line:
                 raise ClusterManagmentError, 'server disconnected'
             if 'Unknown' in line:
                 return 'F'
+            elif 'Invalid credential' in line:       # sometimes qstat simply fails, wait a bit and try again
+                time.sleep(10)
+                return self.control_one_job(id)
