diff --git a/Template/LO/SubProcesses/refine.sh b/Template/LO/SubProcesses/refine.sh
index 577461f..756db37 100644
--- a/Template/LO/SubProcesses/refine.sh
+++ b/Template/LO/SubProcesses/refine.sh
@@ -12,6 +12,10 @@ if [ $SRT_LD_LIBRARY_PATH_SCRAMRT ]; then
   fi
 fi
 
+# If TMPDIR is unset, set it to the condor scratch area if present
+# and fallback to /tmp
+export TMPDIR=${TMPDIR:-${_CONDOR_SCRATCH_DIR:-/tmp}}
+
 if [[ -e MadLoop5_resources.tar.gz && ! -e MadLoop5_resources ]]; then
 tar -xzf MadLoop5_resources.tar.gz
 fi
@@ -72,9 +76,12 @@ j=%(directory)s
 	 rm results.dat
 	 echo "ERROR DETECTED"
 	 echo "end-code not correct $status_code" > results.dat
+         echo "+ Printing job environment:" >> results.dat
+         env >> results.dat
      fi     
      if [[ -e ftn26 ]]; then
          cp ftn26 ftn25
      fi
      cd ../
 
+exit $status_code
diff --git a/Template/LO/SubProcesses/refine_splitted.sh b/Template/LO/SubProcesses/refine_splitted.sh
index d7d5df6..b4b9e57 100644
--- a/Template/LO/SubProcesses/refine_splitted.sh
+++ b/Template/LO/SubProcesses/refine_splitted.sh
@@ -12,6 +12,10 @@ if [ $SRT_LD_LIBRARY_PATH_SCRAMRT ]; then
   fi
 fi
 
+# If TMPDIR is unset, set it to the condor scratch area if present
+# and fallback to /tmp
+export TMPDIR=${TMPDIR:-${_CONDOR_SCRATCH_DIR:-/tmp}}
+
 if [[ -e MadLoop5_resources.tar.gz && ! -e MadLoop5_resources ]]; then
 tar -xzf MadLoop5_resources.tar.gz
 fi
@@ -82,6 +86,9 @@ if [[ $status_code -ne 0 ]]; then
 	 rm results.dat
 	 echo "ERROR DETECTED"
 	 echo "end code not correct $status_code" > results.dat
+         echo "+ Printing job environment:" >> results.dat
+         env >> results.dat
 fi
 
 cd ../
+exit $status_code
diff --git a/Template/LO/SubProcesses/survey.sh b/Template/LO/SubProcesses/survey.sh
index c9ef2c5..ef9ef53 100755
--- a/Template/LO/SubProcesses/survey.sh
+++ b/Template/LO/SubProcesses/survey.sh
@@ -12,6 +12,10 @@ if [ $SRT_LD_LIBRARY_PATH_SCRAMRT ]; then
   fi
 fi
 
+# If TMPDIR is unset, set it to the condor scratch area if present
+# and fallback to /tmp
+export TMPDIR=${TMPDIR:-${_CONDOR_SCRATCH_DIR:-/tmp}}
+
 if [[ -e MadLoop5_resources.tar.gz && ! -e MadLoop5_resources ]]; then
 tar -xzf MadLoop5_resources.tar.gz;
 fi
@@ -80,6 +84,10 @@ for i in $@ ; do
 	 rm results.dat
 	 echo "ERROR DETECTED"
 	 echo "end code not correct $status_code" > results.dat
+         echo "+ Printing job environment:" >> results.dat
+         env >> results.dat
+         cd ../
+         exit $status_code
      fi
      cd ../;
 
@@ -87,6 +95,3 @@ for i in $@ ; do
 done;
 
 # Cleaning 
-
-
-
diff --git a/Template/NLO/SubProcesses/ajob_template b/Template/NLO/SubProcesses/ajob_template
index db5444b..4ded855 100755
--- a/Template/NLO/SubProcesses/ajob_template
+++ b/Template/NLO/SubProcesses/ajob_template
@@ -19,6 +19,10 @@ if [ $SRT_LD_LIBRARY_PATH_SCRAMRT ]; then
   fi
 fi
 
+# If TMPDIR is unset, set it to the condor scratch area if present
+# and fallback to /tmp
+export TMPDIR=${TMPDIR:-${_CONDOR_SCRATCH_DIR:-/tmp}}
+
 tarCounter=0
 while [[ (-f MadLoop5_resources.tar.gz) && (! -f MadLoop5_resources/HelConfigs.dat) && ($tarCounter < 10) ]]; do
     if [[ $tarCounter > 0 ]]; then
@@ -100,6 +104,11 @@ TAGTAGTAGTAGTAGTAGTAG for i in 1 ; do
         ../madevent_mintMC > log.txt <input_app.txt 2>&1
     fi
     status=$?
+    if [[ $status_code -ne 0 ]]; then
+      echo "Error: Status code $status_code" >> log.txt
+      echo "+ Printing environment" >> log.txt
+      env >> log.txt
+    fi
     T="$(($(date +%s)-T))"
     echo "Time in seconds: ${T}" >>log.txt
     cp -f log.txt log_MINT$integration_step.txt  >/dev/null 2>&1
diff --git a/Template/NLO/SubProcesses/reweight_xsec_events.local b/Template/NLO/SubProcesses/reweight_xsec_events.local
index a38c4e8..c8c856b 100644
--- a/Template/NLO/SubProcesses/reweight_xsec_events.local
+++ b/Template/NLO/SubProcesses/reweight_xsec_events.local
@@ -15,6 +15,10 @@ if [ $SRT_LD_LIBRARY_PATH_SCRAMRT ]; then
   fi
 fi
 
+# If TMPDIR is unset, set it to the condor scratch area if present
+# and fallback to /tmp
+export TMPDIR=${TMPDIR:-${_CONDOR_SCRATCH_DIR:-/tmp}}
+
 if [[ -e ./reweight_xsec_events ]]
 then
     (echo $event_file; echo $save_wgts) | ./reweight_xsec_events > reweight_xsec_events.output
