--- a/madgraph/various/misc.py
+++ b/madgraph/various/misc.py
@@ -475,6 +475,7 @@ def copytree(*args, **opts):
 #===============================================================================
 # Compiler which returns smart output error in case of trouble
 #===============================================================================
+@multiple_try()
 def compile(arg=[], cwd=None, mode='fortran', job_specs = True, nb_core=1 ,**opt):
     """compile a given directory"""
 
