--- a/madgraph/interface/amcatnlo_run_interface.py	2019-02-04 12:19:42.000000000 +0100
+++ b/madgraph/interface/amcatnlo_run_interface.py	2019-04-30 10:59:55.718456046 +0200
@@ -2942,6 +2942,6 @@
                 # check for PLUGIN format
                 cluster_class = misc.from_plugin_import(self.plugin_path, 
                                             'new_cluster', cluster_name,
-                                            info = 'cluster handling will be done with PLUGIN: %{plug}s' )
+                                            info = 'cluster handling will be done with PLUGIN: %(plug)s' )
                 if cluster_class:
                     self.cluster = cluster_class(**self.options)

--- a/madgraph/interface/common_run_interface.py        2017-12-13 05:19:14.000000000 +0800
+++ b/madgraph/interface/common_run_interface.py        2019-06-05 08:22:13.536090538 +0800
@@ -3120,7 +3120,7 @@
                 # check for PLUGIN format
                 cluster_class = misc.from_plugin_import(self.plugin_path, 
                                             'new_cluster', cluster_name,
-                                            info = 'cluster handling will be done with PLUGIN: %{plug}s' )
+                                            info = 'cluster handling will be done with PLUGIN: %(plug)s' )
                 if cluster_class:
                     self.cluster = cluster_class(**self.options)
                 else:
@@ -3416,6 +3416,13 @@
 
         # Configure the way to open a file:
         misc.open_file.configure(self.options)
+        # update the path to the PLUGIN directory of MG%
+        if MADEVENT and 'mg5_path' in self.options and self.options['mg5_path']:
+            mg5dir = self.options['mg5_path']
+            if mg5dir not in sys.path:
+                sys.path.append(mg5dir)
+            if pjoin(mg5dir, 'PLUGIN') not in self.plugin_path:
+                self.plugin_path.append(pjoin(mg5dir,'PLUGIN'))
         self.configure_run_mode(self.options['run_mode'])
         return self.options
 
