--- a/madgraph/interface/reweight_interface.py  2019-07-28 20:33:50.606009030 +0200
+++ b/madgraph/interface/reweight_interface.py  2019-07-28 20:49:44.882060940 +0200
@@ -24,6 +24,7 @@
 import tempfile
 import time
 import subprocess
+import fileinput
 from subprocess import Popen, PIPE, STDOUT


@@ -1670,9 +1671,15 @@
             else:
                 nb_core = 1
             os.environ['MENUM'] = '2'
+            os.chdir(pdir)
+            for line in fileinput.FileInput("../Source/make_opts", inplace=1):
+                line=line.replace("DEFAULT_F2PY_COMPILER=f2py","DEFAULT_F2PY_COMPILER=f2py2.7")
+                print line
+#            os.system("make allmatrix2py.so")
             misc.compile(['allmatrix2py.so'], cwd=pdir, nb_core=nb_core)
             if not (self.second_model or self.second_process or self.dedicated_path):
                 os.environ['MENUM'] = '3'
+#                os.system("make allmatrix3py.so")
                 misc.compile(['allmatrix3py.so'], cwd=pdir, nb_core=nb_core)

     def load_module(self, metag=1):

~                                                                                                    
