--- a/MadSpin/decay.py	2019-02-04 12:19:42.000000000 +0100
+++ b/MadSpin/decay.py	2019-12-04 18:19:58.386629000 +0100
@@ -3921,7 +3921,8 @@
                 
         self.cross = 0
         self.error = 0
-        if 'init' in self.banner and (eff!=1 or not any(v==-1 for v in self.br_per_id.values())):
+        if 'init' in self.banner and (eff!=1 or not any(v==-1 for v in self.br_per_id.values())) \
+                                 and not self.options['onlyhelicity']:
             new_init =''
             curr_proc = 0
             has_missing=False
