diff --git a/MG5_aMC_v2_9_13/madgraph/interface/amcatnlo_run_interface.py b/MG5_aMC_v2_9_14/madgraph/interface/amcatnlo_run_interface.py
index f8b611b..2aff879 100755
--- a/madgraph/interface/amcatnlo_run_interface.py
+++ b/madgraph/interface/amcatnlo_run_interface.py
@@ -3910,8 +3910,8 @@ RESTART = %(mint_mode)s
             arg_list = [[shower, out_id, self.run_name]]
 
         self.run_all({rundir: 'shower.sh'}, arg_list, 'shower')
-        self.njobs = 1
-        self.wait_for_complete('shower')
+        #self.njobs = 1
+        #self.wait_for_complete('shower')
 
         # now collect the results
         message = ''
@@ -4739,7 +4739,8 @@ RESTART = %(mint_mode)s
                 [(cwd, exe)] = list(job_dict.items())
                 self.run_exe(exe, args, run_type, cwd)
         
-        self.wait_for_complete(run_type)
+        if self.njobs:
+            self.wait_for_complete(run_type)
 
 
 
diff --git a/MG5_aMC_v2_9_13/madgraph/various/cluster.py b/MG5_aMC_v2_9_14/madgraph/various/cluster.py
index c318415..8be527e 100755
--- a/madgraph/various/cluster.py
+++ b/madgraph/various/cluster.py
@@ -28,13 +28,19 @@ from six.moves import input
 logger = logging.getLogger('madgraph.cluster') 
 
 try:
-    from madgraph import MadGraph5Error
+    from madgraph import MadGraph5Error, MG5DIR
     import madgraph.various.misc as misc
+    MADEVENT=False
 except Exception as error:
     if __debug__:
         print(str(error))
     from internal import MadGraph5Error
     import internal.misc as misc
+    MADEVENT=True
+    LOCALDIR = os.path.realpath(os.path.join(os.path.dirname(__file__), os.path.pardir,
+                                                                os.path.pardir))
+
+
 
 pjoin = os.path.join
    
@@ -752,7 +758,7 @@ class MultiCore(Cluster):
                             % {'pid':pid} )
             out = os.system('kill -15 %(pid)s > /dev/null 2>&1' % {'pid':pid} )   
             
-        self.demons.clear()               
+        self.demons[:] = [] 
 
 
     def wait(self, me_dir, update_status, update_first=None):
@@ -837,9 +843,10 @@ class MultiCore(Cluster):
                     raise self.fail_msg
                 elif isinstance(self.fail_msg, str):
                     raise Exception(self.fail_msg)
-                else:
+                elif self.fail_msg:
                     misc.sprint(self.fail_msg)
                     six.reraise(self.fail_msg[0], self.fail_msg[1], self.fail_msg[2])
+                #else can happen if nothing was submitted
             # reset variable for next submission
             try:
                 self.lock.clear()
@@ -1683,8 +1690,14 @@ class SLURMCluster(Cluster):
         """Submit a job prog to a SLURM cluster"""
         
         me_dir = self.get_jobs_identifier(cwd, prog)
-        
-        
+        import sys
+        if prog == sys.executable:
+            argument.insert(0, prog)
+            if MADEVENT:
+                prog = pjoin(LOCALDIR,'bin','internal','eval.sh') 
+            else:
+                prog = pjoin(MG5DIR, 'Template','Common','bin','internal','eval.sh')
+
         if cwd is None:
             cwd = os.getcwd()
         if stdout is None:
@@ -1700,10 +1713,13 @@ class SLURMCluster(Cluster):
                    '-J', me_dir, 
                    '-e', stderr, prog] + argument
 
+
+
         if self.cluster_queue and self.cluster_queue != 'None':
                 command.insert(1, '-p')
                 command.insert(2, self.cluster_queue)
 
+
         a = misc.Popen(command, stdout=subprocess.PIPE, 
                                       stderr=subprocess.STDOUT,
                                       stdin=subprocess.PIPE, cwd=cwd)
