--- a/madgraph/various/systematics.py
+++ b/madgraph/various/systematics.py
@@ -96,9 +96,16 @@
         self.banner = banner_mod.Banner(self.input.banner)  
         self.force_write_banner = bool(write_banner)
         self.orig_dyn = self.banner.get('run_card', 'dynamical_scale_choice')
-        scalefact = self.banner.get('run_card', 'scalefact')
-        if scalefact != 1:
-            self.orig_dyn = -1
+        if  self.banner.run_card.LO:
+            scalefact = self.banner.get('run_card', 'scalefact')
+            if scalefact != 1:
+                self.orig_dyn = -1
+        else:
+            over1 = self.banner.get('run_card', 'mur_over_ref')
+            over2 = self.banner.get('run_card', 'muf_over_ref')
+            if over1 != 1 or over2 !=1:
+                self.orig_dyn = -1
+
         self.orig_pdf = self.banner.run_card.get_lhapdf_id()
         matching_mode = self.banner.get('run_card', 'ickkw')
 
@@ -141,7 +148,7 @@
             isEVA=True
             pdf='0'
         # eva-on-DIS(lhapdf)
-        elif self.banner.run_card['pdlabel1']=='eva' and self.banner.run_card['pdlabel2']=='lhapdf':
+        elif self.banner.run_card.LO and (self.banner.run_card['pdlabel1']=='eva' and self.banner.run_card['pdlabel2']=='lhapdf'):
             if abs(beam1) == 11 or abs(beam1) == 13:
                 self.b1 = beam1
             else:
@@ -149,7 +156,7 @@
             #self.b2 = beam2//2212
             isEVAxDIS=True
         # DIS(lhapdf)-on-eva
-        elif self.banner.run_card['pdlabel1']=='lhapdf' and self.banner.run_card['pdlabel2']=='eva':
+        elif self.banner.run_card.LO and (self.banner.run_card['pdlabel1']=='lhapdf' and self.banner.run_card['pdlabel2']=='eva'):
             if abs(beam2) == 11 or abs(beam2) == 13:
                 self.b2 = beam2
             else:
@@ -1083,7 +1090,7 @@
                 
                 
                 if __debug__ and dyn== -1 and Dmur==1 and Dmuf==1 and pdf==self.orig_pdf:
-                    if not misc.equal(tmp, onewgt.ref_wgt, sig_fig=2):
+                    if not misc.equal(tmp, onewgt.ref_wgt, sig_fig=1):
                         misc.sprint(tmp, onewgt.ref_wgt, (tmp-onewgt.ref_wgt)/tmp)
                         misc.sprint(onewgt)
                         misc.sprint(cevent)

--- a/madgraph/various/lhe_parser.py
+++ b/madgraph/various/lhe_parser.py
@@ -2670,7 +2670,7 @@
             bjks : %(bjks)s
             scales**2, gs: %(scales2)s %(gs)s
             born/real related : %(born_related)s %(real_related)s
-            type / nfks : %(type)s  %(nfks)s
+            type / nfks : %(orderflag)s %(type)s  %(nfks)s
             to merge : %(to_merge_pdg)s in %(merge_new_pdg)s
             ref_wgt :  %(ref_wgt)s""" % self.__dict__
             return out
@@ -2693,6 +2693,7 @@
             to_add('%.10e', self.real)
             to_add('%i', self.nexternal)
             to_add('%i', self.pdgs)
+            to_add('%i', self.orderflag)
             to_add('%i', self.qcdpower)
             to_add('%.10e', self.bjks)
             to_add('%.10e', self.scales2)
@@ -2744,12 +2745,12 @@
         #    from example: 21 2 -11 12 1 2
         self.pdgs = [int(i) for i in data[6:6+self.nexternal]]
         flag = 6+self.nexternal # new starting point for the position
+        # 5[pre] next integer is the expansion order defined at NLO (from example 404)
+        # New since 3.1.0.
+        self.orderflag = int(data[flag])
         # 5. next integer is the power of g_strong in the matrix elements (as before)
         #    from example: 2
-        self.qcdpower = int(data[flag])
-        # 5[bis] next integer is the expansion order defined at NLO (from example 404)
-        # New since 3.1.0.
-        self.orderflag = int(data[flag+1])
+        self.qcdpower = int(data[flag+1])
         flag= flag+1
         # 6. 2 doubles: The bjorken x's used for this contribution (as before)
         #    from example: 0.52500539D-02 0.30205908D+00 

