--- a/madgraph/interface/madgraph_interface.py	2019-02-04 12:19:42.000000000 +0100
+++ b/madgraph/interface/madgraph_interface_patched.py	2022-06-07 16:49:04.120519000 +0200
@@ -5465,6 +5465,8 @@
                 scheme = 5
                 
         if scheme in [4,5]:
+            self.optimize_order(multi)
+            self._multiparticles[qcd_container] = multi
             logger.warning("Pass the definition of \'j\' and \'p\' to %s flavour scheme." % scheme)
             for container in ['p', 'j']:
                 if container in defined_multiparticles:
