diff --git a/Template/LO/SubProcesses/myamp.f b/Template/LO/SubProcesses/myamp.f
index 7b4b788..f5574a3 100644
--- a/Template/LO/SubProcesses/myamp.f
+++ b/Template/LO/SubProcesses/myamp.f
@@ -364,6 +365,9 @@ c     Look for identical particles to map radiation processes
 
 c     Start loop over propagators
       do i=-1,-(nexternal-3),-1
+         if (i.eq.-(nexternal-3)) then
+              cycle
+         endif
          if (iforest(1,i,iconfig) .eq. 1.or.iforest(1,i,iconfig) .eq. 2)then
               tsgn=-1d0
          endif
