--- a/MadSpin/decay.py
+++ b/MadSpin/decay.py
@@ -2181,9 +2181,9 @@ class decay_all_events(object):
         #self.all_ME, bkp2 = None , self.all_ME    
         #self.width_estimator, bkp3 = self.width_estimator, self.width_estimator  
         modelpath = model.get('modelpath')
-        for me in self.all_ME:
-            for d in self.all_ME[me]['decays']:
-                d['decay_struct'] = str(d['decay_struct'])
+        #for me in self.all_ME:
+        #    for d in self.all_ME[me]['decays']:
+        #        d['decay_struct'] = str(d['decay_struct'])
         self.modelpath = modelpath
 
                 
--- a/MadSpin/interface_madspin.py
+++ b/MadSpin/interface_madspin.py
@@ -713,9 +713,9 @@ class MadSpinInterface(extended_cmd.Cmd):
         
         #restore data passed to string to help pickle
         generate_all.all_decay = eval(generate_all.all_decay)
-        for me in generate_all.all_ME:
-            for d in generate_all.all_ME[me]['decays']:
-                d['decay_struct'] = eval(d['decay_struct'])
+        #for me in generate_all.all_ME:
+        #    for d in generate_all.all_ME[me]['decays']:
+        #        d['decay_struct'] = eval(d['decay_struct'])
 
 
         # Re-create information which are not save in the pickle.
