diff -ruN MG5_aMC_v3_5_6/madgraph/interface/madevent_interface.py MG5_aMC_v3_5_6-patch/madgraph/interface/madevent_interface.py
--- a/madgraph/interface/madevent_interface.py	2024-09-27 05:19:41.000000000 +0900
+++ b/madgraph/interface/madevent_interface.py	2025-05-06 13:12:11.000000000 +0900
@@ -7056,6 +7056,7 @@
         
         for data in partials_info:
             AllEvent.add(*data)
+            sum_xsec += data[1]
 
         if not hasattr(self,'proc_characteristic'):
             self.proc_characteristic = self.get_characteristics()
diff -ruN MG5_aMC_v3_5_6/madgraph/various/lhe_parser.py MG5_aMC_v3_5_6-patch/madgraph/various/lhe_parser.py
--- a/madgraph/various/lhe_parser.py	2024-09-27 05:19:42.000000000 +0900
+++ b/madgraph/various/lhe_parser.py	2025-05-06 13:12:32.000000000 +0900
@@ -1025,12 +1025,12 @@
                 from_init = True
 
             if not from_init:
-                if group in grouped_cross:
-                    grouped_cross[group] += self.allcross[i]
-                    grouped_error[group] += self.error[i]**2 
+                if int(group) in grouped_cross:
+                    grouped_cross[int(group)] += self.allcross[i]
+                    grouped_error[int(group)] += self.error[i]**2 
                 else:
-                    grouped_cross[group] = self.allcross[i]
-                    grouped_error[group] = self.error[i]**2
+                    grouped_cross[int(group)] = self.allcross[i]
+                    grouped_error[int(group)] = self.error[i]**2
             else:
                 ban = banner_mod.Banner(ff.banner)
                 for line in  ban['init'].split('\n'):
@@ -1038,11 +1038,11 @@
                     if len(splitline)==4:
                         cross, error, _, group = splitline
                         if int(group) in grouped_cross:
-                            grouped_cross[group] += float(cross)
-                            grouped_error[group] += float(error)**2                        
+                            grouped_cross[int(group)] += float(cross)
+                            grouped_error[int(group)] += float(error)**2                        
                         else:
-                            grouped_cross[group] = float(cross)
-                            grouped_error[group] = float(error)**2                             
+                            grouped_cross[int(group)] = float(cross)
+                            grouped_error[int(group)] = float(error)**2                             
         nb_group = len(grouped_cross)
         
         # compute the information for the first line 
