diff --git a/madgraph/various/cluster.py b/madgraph/various/cluster.py
--- a/madgraph/various/cluster.py
+++ b/madgraph/various/cluster.py
@@ -1220,1 +1220,1 @@
-    maximum_submited_jobs = 2500
+    maximum_submited_jobs = 1000
@@ -1248,2 +1248,3 @@
+        text += "source $VO_CMS_SW_DIR/cmsset_default.sh;eval `scram runtime -sh`;"
         if not os.path.isabs(prog):
             text += "./%s" % prog
