diff -ur ../old/MG5_aMC_v2_3_2_2/madgraph/various/cluster.py MG5_aMC_v2_3_2_2/madgraph/various/cluster.py
--- ../old/MG5_aMC_v2_3_2_2/madgraph/various/cluster.py	2015-09-07 20:07:39.000000001 +0200
+++ MG5_aMC_v2_3_2_2/madgraph/various/cluster.py	2015-09-24 19:47:41.000000001 +0200
@@ -19,6 +19,11 @@
 import glob
 import inspect
 import sys
+import platform
+import signal
+import uuid
+import socket
+import atexit
 
 logger = logging.getLogger('madgraph.cluster') 
 
@@ -83,6 +88,10 @@
     else:
         return True
 
+def cleansubproc(subproc):
+    #print "killing %s" % subproc
+    subproc.terminate()
+
 class Cluster(object):
     """Basic Class for all cluster type submission"""
     name = 'mother class'
@@ -97,6 +106,7 @@
         self.submitted_dirs = [] #HTCaaS
         self.submitted_exes = [] #HTCaaS
         self.submitted_args = [] #HTCaaS
+        self.hold_msg = ""
 
         if 'cluster_queue' in opts:
             self.cluster_queue = opts['cluster_queue']
@@ -307,7 +317,8 @@
             else:
                 nb_job = idle + run + finish + fail
             if fail:
-                raise ClusterManagmentError('Some Jobs are in a Hold/... state. Please try to investigate or contact the IT team')
+                raise ClusterManagmentError('Some Jobs are in a Hold/... state. Error messages are below.' 
+                        'Please try to investigate or contact the IT team. \n%s' % self.hold_msg)
             if idle + run == 0:
                 #time.sleep(20) #security to ensure that the file are really written on the disk
                 logger.info('All jobs finished')
@@ -840,6 +851,7 @@
                   Universe = vanilla
                   notification = Error
                   Initialdir = %(cwd)s
+                  Request_memory = 528
                   %(requirement)s
                   getenv=True
                   queue 1
@@ -916,6 +928,7 @@
                   Universe = vanilla
                   notification = Error
                   Initialdir = %(cwd)s
+                  Request_memory = 528
                   %(requirement)s
                   getenv=True
                   queue 1
@@ -1023,6 +1036,13 @@
                     idle += 1
                 elif status == 'R':
                     run += 1
+                elif status == 'H':
+                    error = misc.Popen(["condor_q", "-format", "'%s\n'", "HoldReason", id], 
+                                            stdout=subprocess.PIPE)
+                    self.hold_msg += "Hold message for ID %s:" % id
+                    for line in error.stdout:
+                        self.hold_msg += line
+                    fail += 1
                 elif status != 'C':
                     fail += 1
 
@@ -1349,6 +1358,80 @@
     name = 'lsf'
     job_id = 'LSB_JOBID'
 
+    def __init__(self,*args, **opts):
+        """Init the cluster"""
+        Cluster.__init__(self,*args, **opts)
+
+        if self.temp_dir!=None:
+            self.dorsync = True
+            #print "starting rsync"
+          
+            cwd = os.getcwd()
+
+            self.rsyncroot = cwd
+            
+            self.rsyncmodule = str(uuid.uuid4())
+            
+            #get free port number for rsyncd
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.bind(('localhost', 0))
+            addr, port = sock.getsockname()
+            sock.close()    
+            
+            self.rsyncport = port
+            #print self.rsyncport
+            
+            rsynclog = os.path.join(cwd, 'rsyncd_%i.log' % self.rsyncport)
+            rsynclock = os.path.join(cwd, 'rsyncd_%i.lock' % self.rsyncport)
+            rsyncpid = os.path.join(cwd, 'rsyncd_%i.pid' % self.rsyncport)
+
+            rsyncpasswd = str(uuid.uuid4())
+            
+            self.rsyncuser = 'madgraph'
+            
+            rsyncsecrets = "%s:%s" % (self.rsyncuser,rsyncpasswd)
+            rsyncsecretsfile = os.path.join(cwd, 'rsyncsecrets_%i' % self.rsyncport)
+            secretsh = open(rsyncsecretsfile,'w')
+            os.chmod(rsyncsecretsfile, 0600)
+            secretsh.write(rsyncsecrets)
+          
+            os.environ["MADGRAPHRSYNCPASSWD_%i" % self.rsyncport] = rsyncpasswd
+            #print rsyncpasswd
+
+            rsyncconf = """
+              port = %(rsyncport)s
+              pid file = %(rsyncpid)s
+              log file = %(rsynclog)s
+             
+                [%(rsyncmodule)s]
+                  comment = Random things available for download
+                  lock file = %(rsynclock)s
+                  secrets file = %(rsyncsecrets)s
+                  path = %(path)s
+                  list = yes 
+                  use chroot = no
+                  munge symlinks = no
+                  read only = no
+                  auth users = %(rsyncuser)s
+            """ % {'rsyncport': self.rsyncport,
+                   'rsyncmodule': self.rsyncmodule,
+                   'path': cwd,
+                  'rsynclog' : rsynclog,
+                  'rsynclock' : rsynclock,
+                  'rsyncpid' : rsyncpid,
+                  'rsyncsecrets' : rsyncsecretsfile,
+                  'rsyncuser' : self.rsyncuser,
+                  }
+            
+            rsyncconffile = os.path.join(cwd, 'rsyncd_%i.conf' % self.rsyncport)
+            open(rsyncconffile,'w').write(rsyncconf)
+            
+            self.rsyncd = subprocess.Popen(['rsync','--daemon', '--no-detach', '--config=%s' % rsyncconffile],cwd=cwd,stdout=subprocess.PIPE,stdin=subprocess.PIPE,stderr=subprocess.PIPE)
+            atexit.register(cleansubproc,self.rsyncd)
+            
+        else:
+            self.dorsync = False
+
     @multiple_try()
     def submit(self, prog, argument=[], cwd=None, stdout=None, stderr=None, log=None,
                required_output=[], nb_submit=0):
@@ -1372,6 +1455,8 @@
         if log is None:
             log = '/dev/null'
         
+        text += 'if [ -n $CMSSW_BASE ]; then cd $CMSSW_BASE; eval `scramv1 runtime -sh`; cd -; fi;'
+        
         text += prog
         if argument:
             text += ' ' + ' '.join(argument)
@@ -1397,7 +1482,137 @@
         self.submitted_ids.append(id)
         return id        
         
+    @store_input()
+    @multiple_try()
+    def submit2(self, prog, argument=[], cwd=None, stdout=None, stderr=None,
+            log=None, input_files=[], output_files=[], required_output=[],nb_submit=0):
+        """How to make one submission. Return status id on the cluster.
+        NO SHARE DISK"""
+
+        #print "running lsf submit2"
+
+        if cwd is None:
+            cwd = os.getcwd()
+        if not os.path.exists(prog):
+            prog = os.path.join(cwd, prog)
+
+        if not required_output and output_files:
+            required_output = output_files
+
+        if not self.dorsync or (not input_files and not output_files):
+            # not input/output so not using submit2
+            return self.submit(prog, argument, cwd, stdout, stderr, log,
+        required_output=required_output, nb_submit=nb_submit)
+
+        if self.rsyncd.poll()!=None:
+            raise RuntimeError("rsyncd not running")
+
+        if cwd is None:
+            cwd = os.getcwd()
+        if not os.path.exists(prog):
+            prog = os.path.join(cwd, prog)
+        temp_file_name = "sub." + os.path.basename(prog) + '.'.join(argument)
+               
+        input_files.append(prog)                
+               
+        hostname = platform.node()
+               
+        rsynccwd = cwd
+        if rsynccwd.startswith(self.rsyncroot):
+            rsynccwd = rsynccwd[len(self.rsyncroot):]                   
+               
+        infilelist = ""
+        for input_file in input_files:
+            #make sure input_files are absolute paths
+            if not input_file.startswith('/'):
+                input_file = os.path.join(cwd,input_file)
+            #convert to paths relative to rsyncd root
+            if input_file.startswith(self.rsyncroot):
+                input_file = input_file[len(self.rsyncroot):]
+            infilelist += "%s@%s::%s/%s " % (self.rsyncuser,hostname,self.rsyncmodule, input_file)
+        infilelist += "./"
+        
+        outfilelist = ""
+        for output_file in output_files:
+            outfilelist += "%s " % (output_file)
+        outfilelist += "%s@%s::%s/%s" % (self.rsyncuser,hostname,self.rsyncmodule,rsynccwd)
+            
+        text = """#!/bin/bash
         
+            SUBMITTERHOST=%(hostname)s            
+
+            if [ -n $CMSSW_VERSION ]
+            then
+              scramv1 project CMSSW $CMSSW_VERSION
+              cd $CMSSW_VERSION
+              eval `scramv1 runtime -sh`
+              cd -
+            fi
+                             
+            export RSYNC_PASSWORD=$MADGRAPHRSYNCPASSWD_%(rsyncport)s
+                 
+            #dereference symlinks for input
+            rsync -vvv --timeout=600 --contimeout=600 --port %(rsyncport)s -rptL %(infilelist)s
+
+            echo '%(arguments)s' > arguments
+            chmod +x ./%(script)s        
+            %(program)s ./%(script)s %(arguments)s
+            
+            #copy symlinks as symlinks for output and don't overwrite existing files unless updated
+            rsync -vvv --timeout=600 --contimeout=600 --port %(rsyncport)s -rptul %(outfilelist)s
+            
+            """
+        dico = {'script': os.path.basename(prog),
+        'hostname': hostname,
+        'infilelist': infilelist,
+        'outfilelist': outfilelist,
+        'output_files': ' '.join(output_files),
+        'rsyncport': self.rsyncport,
+        'arguments': ' '.join([str(a) for a in argument]),
+        'program': ' ' if '.py' in prog else 'bash'}
+
+        me_dir = self.get_jobs_identifier(cwd, prog)
+
+        text = text % dico
+        cwdpath = "/tmp/" + os.environ.get("USER", '')
+        command = ['bsub', '-cwd', cwdpath, '-C0', '-J', me_dir]
+        if cwd is None:
+            cwd = os.getcwd()
+        #else:
+            #text += " cd %s;" % cwd
+        if stdout and isinstance(stdout, str):
+            command.extend(['-o', stdout])
+        if stderr and isinstance(stdout, str):
+            command.extend(['-e', stderr])
+        elif stderr == -2: # -2 is subprocess.STDOUT
+            pass
+        if log is None:
+            log = '/dev/null'
+
+        if self.cluster_queue and self.cluster_queue != 'None':
+            command.extend(['-q', self.cluster_queue])
+
+        submitenv = os.environ.copy()
+        submitenv["TMPDIR"] = "/tmp/" + submitenv.get("USER", '')
+        a = misc.Popen(command, stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        stdin=subprocess.PIPE, cwd=cwd,
+        env=submitenv)
+
+        output = a.communicate(text)[0]
+        #Job <nnnn> is submitted to default queue <normal>.
+        try:
+            id = output.split('>',1)[0].split('<')[1]
+        except:
+            raise ClusterManagmentError, 'fail to submit to the cluster: \n%s' \
+            % output
+        if not id.isdigit():
+            raise ClusterManagmentError, 'fail to submit to the cluster: \n%s' \
+            % output
+        self.submitted += 1
+        self.submitted_ids.append(id)
+        return id         
+                
     @multiple_try()
     def control_one_job(self, id):
         """ control the status of a single job with it's cluster id """
diff -ur ../old/MG5_aMC_v2_3_2_2/MadSpin/decay.py MG5_aMC_v2_3_2_2/MadSpin/decay.py
--- ../old/MG5_aMC_v2_3_2_2/MadSpin/decay.py	2015-09-07 20:07:39.000000001 +0200
+++ MG5_aMC_v2_3_2_2/MadSpin/decay.py	2015-09-24 21:29:32.000000001 +0200
@@ -2043,13 +2043,15 @@
                 pid=abs(int(pidvalue[0]))
                 value=float(pidvalue[1])
                 MC_masses[pid]=value
-                if pid in pid_heavyquarks:
+                if pid in pid_heavyquarks or value==0.:
                     value_ME=self.banner.get('param_card','mass', pid).value
                     if value_ME>1E-10:
                         if pid==5:
                             logger.warning('set the mass of the b-quark to its value in the param_card.dat: %s GeV ' % value_ME)
-                        if pid==4:
+                        elif pid==4:
                             logger.warning('set the mass of the c-quark to its value in the param_card.dat: %s GeV ' % value_ME)
+                        else:
+                            logger.warning('set the mass for pid %i to its value in the param_card.dat: %s GeV ' % (pid,value_ME))
                         MC_masses[pid]=value_ME
             
         return MC_masses 
diff -ur ../old/MG5_aMC_v2_3_2_2/Template/LO/bin/internal/Gridpack/run.sh MG5_aMC_v2_3_2_2/Template/LO/bin/internal/Gridpack/run.sh
--- ../old/MG5_aMC_v2_3_2_2/Template/LO/bin/internal/Gridpack/run.sh	2015-09-07 20:07:39.000000001 +0200
+++ MG5_aMC_v2_3_2_2/Template/LO/bin/internal/Gridpack/run.sh	2015-09-24 19:53:42.000000001 +0200
@@ -78,43 +78,5 @@
     cd ..
 fi
 
-if [[ -e ./DECAY/decay ]]; then
-    cd DECAY
-    echo -$seed > iseed.dat
-    for ((i = 1 ;  i <= 20;  i++)) ; do
-	if [[ -e decay_$i\.in ]]; then
-	    echo "Decaying events..."
-	    mv ../events.lhe ../events_in.lhe
-	    ./decay < decay_$i\.in
-	fi
-    done
-    cd ..
-fi
-
-if [[ -e ./REPLACE/replace.pl ]]; then
-    for ((i = 1 ;  i <= 20;  i++)) ; do
-	if [[ -e ./REPLACE/replace_card$i\.dat ]];then
-	    echo "Adding flavors..."
-	    mv ./events.lhe ./events_in.lhe
-	    cd ./REPLACE
-	    ./replace.pl ../events_in.lhe ../events.lhe < replace_card$i\.dat
-	    cd ..
-	fi
-    done
-fi
-
-# part added by Stephen Mrenna to correct the kinematics of the replaced
-#  particles
-if [[ -e ./madevent/bin/internal/addmasses.py ]]; then
-  mv ./events.lhe ./events.lhe.0
-  python ./madevent/bin/internal/addmasses.py ./events.lhe.0 ./events.lhe
-  if [[ $? -eq 0 ]]; then
-     echo "Mass added"
-     rm -rf ./events.lhe.0 &> /dev/null
-  else
-     mv ./events.lhe.0 ./events.lhe
-  fi
-fi  
-
 gzip -f events.lhe
 exit
diff -ur ../old/MG5_aMC_v2_3_2_2/Template/NLO/SubProcesses/MCmasses_PYTHIA8.inc MG5_aMC_v2_3_2_2/Template/NLO/SubProcesses/MCmasses_PYTHIA8.inc
--- ../old/MG5_aMC_v2_3_2_2/Template/NLO/SubProcesses/MCmasses_PYTHIA8.inc	2015-09-07 20:07:39.000000001 +0200
+++ MG5_aMC_v2_3_2_2/Template/NLO/SubProcesses/MCmasses_PYTHIA8.inc	2015-09-24 20:12:25.000000001 +0200
@@ -1,12 +1,12 @@
-      mcmass(1)=0.33d0
-      mcmass(2)=0.33d0
-      mcmass(3)=0.50d0
-      mcmass(4)=1.50d0
-      mcmass(5)=4.80d0
-      mcmass(11)=0.510998928d-3
-      mcmass(12)=0.d0
-      mcmass(13)=0.1056583715d0
-      mcmass(14)=0.d0
-      mcmass(15)=1.77682d0
-      mcmass(16)=0.d0
+      mcmass(1)=0.0d0
+      mcmass(2)=0.0d0
+      mcmass(3)=0.0d0
+      mcmass(4)=0.0d0
+      mcmass(5)=0.0d0
+      mcmass(11)=0.0d0
+      mcmass(12)=0.0d0
+      mcmass(13)=0.0d0
+      mcmass(14)=0.0d0
+      mcmass(15)=0.0d0
+      mcmass(16)=0.0d0
       mcmass(21)=0.0d0
diff -ur ../old/MG5_aMC_v2_3_2_2/Template/NLO/SubProcesses/setscales.f MG5_aMC_v2_3_2_2/Template/NLO/SubProcesses/setscales.f
--- ../old/MG5_aMC_v2_3_2_2/Template/NLO/SubProcesses/setscales.f	2015-09-07 20:07:39.000000001 +0200
+++ MG5_aMC_v2_3_2_2/Template/NLO/SubProcesses/setscales.f	2015-09-24 19:54:56.000000001 +0200
@@ -411,7 +411,7 @@
       tmp=0
       if(ickkw.eq.3)then
 c FxFx merging scale:
-        tmp=min(FxFx_fac_scale(1),FxFx_fac_scale(2))
+        tmp=(FxFx_fac_scale(1)+FxFx_fac_scale(2))/2d0
         temp_scale_id='FxFx merging scale'
       elseif(imuftype.eq.1)then
         tmp=scale_global_reference(pp)
diff -ur ../old/MG5_aMC_v2_3_2_2/madgraph/interface/madgraph_interface.py MG5_aMC_v2_3_2_2/madgraph/interface/madgraph_interface.py
--- ../old/MG5_aMC_v2_3_2_2/madgraph/interface/madgraph_interface.py       2015-09-07 20:07:39.000000000 +0200
+++ MG5_aMC_v2_3_2_2/madgraph/interface/madgraph_interface.py    2015-11-17 10:50:56.985751642 +0100
@@ -5851,6 +5851,9 @@
         elif args[0] in ['cluster_local_path']:
             self.options[args[0]] = args[1].strip()
 
+        elif args[0] in ['cluster_queue']:
+            self.options[args[0]] = args[1].strip()
+        
         elif args[0] == 'cluster_status_update':
             if '(' in args[1]:
                 data = ' '.join([a for a in args[1:] if not a.startswith('-')])
