diff -ur ../orig/MG5_aMC_v2_3_0/madgraph/interface/common_run_interface.py MG5_aMC_v2_3_0/madgraph/interface/common_run_interface.py
--- ../orig/MG5_aMC_v2_3_0/madgraph/interface/common_run_interface.py	2015-07-02 04:18:10.000000001 +0200
+++ MG5_aMC_v2_3_0/madgraph/interface/common_run_interface.py	2015-07-21 17:42:01.000000001 +0200
@@ -1314,7 +1314,7 @@
     def get_pdf_input_filename(self):
         """return the name of the file which is used by the pdfset"""
 
-        if self.options["cluster_local_path"] and self.options['run_mode'] ==1:
+        if self.options["cluster_local_path"] and os.path.exists(self.options["cluster_local_path"]) and self.options['run_mode'] ==1:
             # no need to transfer the pdf.
             return ''
         
diff -ur ../orig/MG5_aMC_v2_3_0/madgraph/interface/madgraph_interface.py MG5_aMC_v2_3_0/madgraph/interface/madgraph_interface.py
--- ../orig/MG5_aMC_v2_3_0/madgraph/interface/madgraph_interface.py	2015-07-02 04:18:10.000000001 +0200
+++ MG5_aMC_v2_3_0/madgraph/interface/madgraph_interface.py	2015-07-21 01:49:58.000000001 +0200
@@ -6432,7 +6432,7 @@
             logger.warning(warning_text)
             
         if not model:
-            modelname = self._curr_model.get('modelpath')
+            modelname = self._curr_model.get('modelpath+restriction')
             with misc.MuteLogger(['madgraph'], ['INFO']):
                 model = import_ufo.import_model(modelname, decay=True)
         else:
diff -ur ../orig/MG5_aMC_v2_3_0/madgraph/various/cluster.py MG5_aMC_v2_3_0/madgraph/various/cluster.py
--- ../orig/MG5_aMC_v2_3_0/madgraph/various/cluster.py	2015-07-02 04:18:10.000000001 +0200
+++ MG5_aMC_v2_3_0/madgraph/various/cluster.py	2015-07-21 18:03:58.000000001 +0200
@@ -19,6 +19,11 @@
 import glob
 import inspect
 import sys
+import platform
+import signal
+import uuid
+import socket
+import atexit
 
 logger = logging.getLogger('madgraph.cluster') 
 
@@ -83,6 +88,10 @@
     else:
         return True
 
+def cleansubproc(subproc):
+    #print "killing %s" % subproc
+    subproc.terminate()
+
 class Cluster(object):
     """Basic Class for all cluster type submission"""
     name = 'mother class'
@@ -106,7 +115,7 @@
         for key,value in opts.items():
             self.options[key] = value
         self.nb_retry = opts['cluster_nb_retry'] if 'cluster_nb_retry' in opts else 0
-        self.cluster_retry_wait = opts['cluster_retry_wait'] if 'cluster_retry_wait' in opts else 300
+        self.cluster_retry_wait = float(opts['cluster_retry_wait']) if 'cluster_retry_wait' in opts else 300
         self.options = dict(opts)
         self.retry_args = {}
         # controlling jobs in controlled type submision
@@ -1332,6 +1341,80 @@
     name = 'lsf'
     job_id = 'LSB_JOBID'
 
+    def __init__(self,*args, **opts):
+        """Init the cluster"""
+        Cluster.__init__(self,*args, **opts)
+
+        if self.temp_dir!=None:
+            self.dorsync = True
+            #print "starting rsync"
+          
+            cwd = os.getcwd()
+
+            self.rsyncroot = cwd
+            
+            self.rsyncmodule = str(uuid.uuid4())
+            
+            #get free port number for rsyncd
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            sock.bind(('localhost', 0))
+            addr, port = sock.getsockname()
+            sock.close()    
+            
+            self.rsyncport = port
+            #print self.rsyncport
+            
+            rsynclog = os.path.join(cwd, 'rsyncd_%i.log' % self.rsyncport)
+            rsynclock = os.path.join(cwd, 'rsyncd_%i.lock' % self.rsyncport)
+            rsyncpid = os.path.join(cwd, 'rsyncd_%i.pid' % self.rsyncport)
+
+            rsyncpasswd = str(uuid.uuid4())
+            
+            self.rsyncuser = 'madgraph'
+            
+            rsyncsecrets = "%s:%s" % (self.rsyncuser,rsyncpasswd)
+            rsyncsecretsfile = os.path.join(cwd, 'rsyncsecrets_%i' % self.rsyncport)
+            secretsh = open(rsyncsecretsfile,'w')
+            os.chmod(rsyncsecretsfile, 0600)
+            secretsh.write(rsyncsecrets)
+          
+            os.environ["MADGRAPHRSYNCPASSWD_%i" % self.rsyncport] = rsyncpasswd
+            #print rsyncpasswd
+
+            rsyncconf = """
+              port = %(rsyncport)s
+              pid file = %(rsyncpid)s
+              log file = %(rsynclog)s
+             
+                [%(rsyncmodule)s]
+                  comment = Random things available for download
+                  lock file = %(rsynclock)s
+                  secrets file = %(rsyncsecrets)s
+                  path = %(path)s
+                  list = yes 
+                  use chroot = no
+                  munge symlinks = no
+                  read only = no
+                  auth users = %(rsyncuser)s
+            """ % {'rsyncport': self.rsyncport,
+                   'rsyncmodule': self.rsyncmodule,
+                   'path': cwd,
+                  'rsynclog' : rsynclog,
+                  'rsynclock' : rsynclock,
+                  'rsyncpid' : rsyncpid,
+                  'rsyncsecrets' : rsyncsecretsfile,
+                  'rsyncuser' : self.rsyncuser,
+                  }
+            
+            rsyncconffile = os.path.join(cwd, 'rsyncd_%i.conf' % self.rsyncport)
+            open(rsyncconffile,'w').write(rsyncconf)
+            
+            self.rsyncd = subprocess.Popen(['rsync','--daemon', '--no-detach', '--config=%s' % rsyncconffile],cwd=cwd,stdout=subprocess.PIPE,stdin=subprocess.PIPE,stderr=subprocess.PIPE)
+            atexit.register(cleansubproc,self.rsyncd)
+            
+        else:
+            self.dorsync = False
+
     @multiple_try()
     def submit(self, prog, argument=[], cwd=None, stdout=None, stderr=None, log=None,
                required_output=[], nb_submit=0):
@@ -1355,6 +1438,8 @@
         if log is None:
             log = '/dev/null'
         
+        text += 'if [ -n $CMSSW_BASE ]; then cd $CMSSW_BASE; eval `scramv1 runtime -sh`; cd -; fi;'
+        
         text += prog
         if argument:
             text += ' ' + ' '.join(argument)
@@ -1380,6 +1465,136 @@
         self.submitted_ids.append(id)
         return id        
         
+    @store_input()
+    @multiple_try()
+    def submit2(self, prog, argument=[], cwd=None, stdout=None, stderr=None,
+            log=None, input_files=[], output_files=[], required_output=[],nb_submit=0):
+        """How to make one submission. Return status id on the cluster.
+        NO SHARE DISK"""
+
+        #print "running lsf submit2"
+
+        if cwd is None:
+            cwd = os.getcwd()
+        if not os.path.exists(prog):
+            prog = os.path.join(cwd, prog)
+
+        if not required_output and output_files:
+            required_output = output_files
+
+        if not self.dorsync or (not input_files and not output_files):
+            # not input/output so not using submit2
+            return self.submit(prog, argument, cwd, stdout, stderr, log,
+        required_output=required_output, nb_submit=nb_submit)
+
+        if self.rsyncd.poll()!=None:
+            raise RuntimeError("rsyncd not running")
+
+        if cwd is None:
+            cwd = os.getcwd()
+        if not os.path.exists(prog):
+            prog = os.path.join(cwd, prog)
+        temp_file_name = "sub." + os.path.basename(prog) + '.'.join(argument)
+               
+        input_files.append(prog)                
+               
+        hostname = platform.node()
+               
+        rsynccwd = cwd
+        if rsynccwd.startswith(self.rsyncroot):
+            rsynccwd = rsynccwd[len(self.rsyncroot):]                   
+               
+        infilelist = ""
+        for input_file in input_files:
+            #make sure input_files are absolute paths
+            if not input_file.startswith('/'):
+                input_file = os.path.join(cwd,input_file)
+            #convert to paths relative to rsyncd root
+            if input_file.startswith(self.rsyncroot):
+                input_file = input_file[len(self.rsyncroot):]
+            infilelist += "%s@%s::%s/%s " % (self.rsyncuser,hostname,self.rsyncmodule, input_file)
+        infilelist += "./"
+        
+        outfilelist = ""
+        for output_file in output_files:
+            outfilelist += "%s " % (output_file)
+        outfilelist += "%s@%s::%s/%s" % (self.rsyncuser,hostname,self.rsyncmodule,rsynccwd)
+            
+        text = """#!/bin/bash
+        
+            SUBMITTERHOST=%(hostname)s            
+
+            if [ -n $CMSSW_VERSION ]
+            then
+              scramv1 project CMSSW $CMSSW_VERSION
+              cd $CMSSW_VERSION
+              eval `scramv1 runtime -sh`
+              cd -
+            fi
+                             
+            export RSYNC_PASSWORD=$MADGRAPHRSYNCPASSWD_%(rsyncport)s
+                 
+            #dereference symlinks for input
+            rsync -vvv --timeout=600 --contimeout=600 --port %(rsyncport)s -rptL %(infilelist)s
+
+            echo '%(arguments)s' > arguments
+            chmod +x ./%(script)s        
+            %(program)s ./%(script)s %(arguments)s
+            
+            #copy symlinks as symlinks for output and don't overwrite existing files unless updated
+            rsync -vvv --timeout=600 --contimeout=600 --port %(rsyncport)s -rptul %(outfilelist)s
+            
+            """
+        dico = {'script': os.path.basename(prog),
+        'hostname': hostname,
+        'infilelist': infilelist,
+        'outfilelist': outfilelist,
+        'output_files': ' '.join(output_files),
+        'rsyncport': self.rsyncport,
+        'arguments': ' '.join([str(a) for a in argument]),
+        'program': ' ' if '.py' in prog else 'bash'}
+
+        me_dir = self.get_jobs_identifier(cwd, prog)
+
+        text = text % dico
+        cwdpath = "/tmp/" + os.environ.get("USER", '')
+        command = ['bsub', '-cwd', cwdpath, '-C0', '-J', me_dir]
+        if cwd is None:
+            cwd = os.getcwd()
+        #else:
+            #text += " cd %s;" % cwd
+        if stdout and isinstance(stdout, str):
+            command.extend(['-o', stdout])
+        if stderr and isinstance(stdout, str):
+            command.extend(['-e', stderr])
+        elif stderr == -2: # -2 is subprocess.STDOUT
+            pass
+        if log is None:
+            log = '/dev/null'
+
+        if self.cluster_queue and self.cluster_queue != 'None':
+            command.extend(['-q', self.cluster_queue])
+
+        submitenv = os.environ.copy()
+        submitenv["TMPDIR"] = "/tmp/" + submitenv.get("USER", '')
+        a = misc.Popen(command, stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        stdin=subprocess.PIPE, cwd=cwd,
+        env=submitenv)
+
+        output = a.communicate(text)[0]
+        #Job <nnnn> is submitted to default queue <normal>.
+        try:
+            id = output.split('>',1)[0].split('<')[1]
+        except:
+            raise ClusterManagmentError, 'fail to submit to the cluster: \n%s' \
+            % output
+        if not id.isdigit():
+            raise ClusterManagmentError, 'fail to submit to the cluster: \n%s' \
+            % output
+        self.submitted += 1
+        self.submitted_ids.append(id)
+        return id         
         
     @multiple_try()
     def control_one_job(self, id):
diff -ur ../orig/MG5_aMC_v2_3_0/MadSpin/interface_madspin.py MG5_aMC_v2_3_0/MadSpin/interface_madspin.py
--- ../orig/MG5_aMC_v2_3_0/MadSpin/interface_madspin.py	2015-07-02 04:18:10.000000001 +0200
+++ MG5_aMC_v2_3_0/MadSpin/interface_madspin.py	2015-07-07 18:03:38.000000001 +0200
@@ -584,7 +584,10 @@
 
         #replace init information
         generate_all.banner['init'] = self.banner['init']
-        
+        #replace run card if present in header (to make sure correct random seed is recorded in output file)
+        if 'mgruncard' in self.banner:
+            generate_all.banner['mgruncard'] = self.banner['mgruncard']        
+
         # NOW we have all the information available for RUNNING
         
         if self.seed:
diff -ur ../orig/MG5_aMC_v2_3_0/models/check_param_card.py MG5_aMC_v2_3_0/models/check_param_card.py
--- ../orig/MG5_aMC_v2_3_0/models/check_param_card.py	2015-07-02 04:18:10.000000001 +0200
+++ MG5_aMC_v2_3_0/models/check_param_card.py	2015-07-22 02:08:40.000000001 +0200
@@ -978,7 +978,10 @@
     # Td
     yd = card['yd'].get([3, 3]).value
     td = card['td'].get([3, 3]).value
-    card.mod_param('td', [3,3], 'ad', [3,3], value= td/yd, comment='A_b(Q) DRbar')
+    if td:
+        card.mod_param('td', [3,3], 'ad', [3,3], value= td/yd, comment='A_b(Q) DRbar')
+    else:
+        card.mod_param('td', [3,3], 'ad', [3,3], value= 0., comment='A_b(Q) DRbar')
     card.add_param('ad', [1,1], 0, 'A_d(Q) DRbar')
     card.add_param('ad', [2,2], 0, 'A_s(Q) DRbar')
     card['ad'].scale = scale
diff -ur ../orig/MG5_aMC_v2_3_0/models/mssm/restrict_no_b_mass.dat MG5_aMC_v2_3_0/models/mssm/restrict_no_b_mass.dat
--- ../orig/MG5_aMC_v2_3_0/models/mssm/restrict_no_b_mass.dat	2015-07-02 04:18:10.000000001 +0200
+++ MG5_aMC_v2_3_0/models/mssm/restrict_no_b_mass.dat	2015-07-22 02:06:56.000000001 +0200
@@ -235,7 +235,7 @@
 ## INFORMATION FOR TD
 ###################################
 Block TD 
-    3   3 -1.106937e+02 # Rtd33 
+    3   3 0.000000e+00 # Rtd33 
 
 ###################################
 ## INFORMATION FOR TE
diff -ur ../orig/MG5_aMC_v2_3_0/Template/LO/bin/internal/Gridpack/run.sh MG5_aMC_v2_3_0/Template/LO/bin/internal/Gridpack/run.sh
--- ../orig/MG5_aMC_v2_3_0/Template/LO/bin/internal/Gridpack/run.sh	2015-07-02 04:18:10.000000001 +0200
+++ MG5_aMC_v2_3_0/Template/LO/bin/internal/Gridpack/run.sh	2015-07-07 16:53:02.000000001 +0200
@@ -78,43 +78,5 @@
     cd ..
 fi
 
-if [[ -e ./DECAY/decay ]]; then
-    cd DECAY
-    echo -$seed > iseed.dat
-    for ((i = 1 ;  i <= 20;  i++)) ; do
-	if [[ -e decay_$i\.in ]]; then
-	    echo "Decaying events..."
-	    mv ../events.lhe ../events_in.lhe
-	    ./decay < decay_$i\.in
-	fi
-    done
-    cd ..
-fi
-
-if [[ -e ./REPLACE/replace.pl ]]; then
-    for ((i = 1 ;  i <= 20;  i++)) ; do
-	if [[ -e ./REPLACE/replace_card$i\.dat ]];then
-	    echo "Adding flavors..."
-	    mv ./events.lhe ./events_in.lhe
-	    cd ./REPLACE
-	    ./replace.pl ../events_in.lhe ../events.lhe < replace_card$i\.dat
-	    cd ..
-	fi
-    done
-fi
-
-# part added by Stephen Mrenna to correct the kinematics of the replaced
-#  particles
-if [[ -e ./madevent/bin/internal/addmasses.py ]]; then
-  mv ./events.lhe ./events.lhe.0
-  python ./madevent/bin/internal/addmasses.py ./events.lhe.0 ./events.lhe
-  if [[ $? -eq 0 ]]; then
-     echo "Mass added"
-     rm -rf ./events.lhe.0 &> /dev/null
-  else
-     mv ./events.lhe.0 ./events.lhe
-  fi
-fi  
-
 gzip -f events.lhe
 exit
