diff -ur ../orig/MG5_aMC_v2_2_1/madgraph/various/cluster.py ./MG5_aMC_v2_2_1/madgraph/various/cluster.py
--- ../orig/MG5_aMC_v2_2_1/madgraph/various/cluster.py  2014-09-25 16:56:11.000000001 +0200
+++ ./MG5_aMC_v2_2_1/madgraph/various/cluster.py        2014-10-20 19:21:21.000000001 +0200
@@ -1290,7 +1290,7 @@
             me_dir = 'a' + me_dir[1:]
         
         text = ""
-        command = ['bsub', '-J', me_dir]
+        command = ['bsub', '-C0', '-J', me_dir]
         if cwd is None:
             cwd = os.getcwd()
         else: 
@@ -1303,6 +1303,8 @@
             pass
         if log is None:
             log = '/dev/null'
+
+        text += 'if [ -n $CMSSW_BASE ]; then cd $CMSSW_BASE; eval `scramv1 runtime -sh`; cd -; fi;'
         
         text += prog
         if argument:
@@ -1364,7 +1366,7 @@
         cmd = "bjobs " + ' '.join(self.submitted_ids) 
         status = misc.Popen([cmd], shell=True, stdout=subprocess.PIPE)
 
-        idle, run, fail = 0, 0, 0
+        jobstatus = {}
         for line in status.stdout:
             line = line.strip()
             if 'JOBID' in line:
@@ -1374,18 +1376,26 @@
             if id not in self.submitted_ids:
                 continue
             status = splitline[2]
+            jobstatus[id] = status
+
+                
+        idle, run, fail = 0, 0, 0
+        for id in self.submitted_ids:
+            if id in jobstatus:
+                status = jobstatus[id]
+            else:
+                status = 'MISSING'
+                
             if status == 'RUN':
                 run += 1
             elif status == 'PEND':
                 idle += 1
-            elif status == 'DONE':
+            else:
                 status = self.check_termination(id)
                 if status == 'wait':
                     run += 1
                 elif status == 'resubmit':
                     idle += 1
-            else:
-                fail += 1
 
         return idle, run, self.submitted - (idle+run+fail), fail

diff -ur old/MG5_aMC_v2_2_1/Template/LO/bin/internal/Gridpack/run.sh ./MG5_aMC_v2_2_1/Template/LO/bin/internal/Gridpack/run.sh
--- old/MG5_aMC_v2_2_1/Template/LO/bin/internal/Gridpack/run.sh 2014-09-25 16:56:11.000000000 +0200
+++ ./MG5_aMC_v2_2_1/Template/LO/bin/internal/Gridpack/run.sh   2014-09-26 02:14:50.115432890 +0200
@@ -78,37 +78,5 @@
     cd ..
 fi
 
-if [[ -e ./DECAY/decay ]]; then
-    cd DECAY
-    echo -$seed > iseed.dat
-    for ((i = 1 ;  i <= 20;  i++)) ; do
-       if [[ -e decay_$i\.in ]]; then
-           echo "Decaying events..."
-           mv ../events.lhe ../events_in.lhe
-           ./decay < decay_$i\.in
-       fi
-    done
-    cd ..
-fi
-
-if [[ -e ./REPLACE/replace.pl ]]; then
-    for ((i = 1 ;  i <= 20;  i++)) ; do
-       if [[ -e ./REPLACE/replace_card$i\.dat ]];then
-           echo "Adding flavors..."
-           mv ./events.lhe ./events_in.lhe
-           cd ./REPLACE
-           ./replace.pl ../events_in.lhe ../events.lhe < replace_card$i\.dat
-           cd ..
-       fi
-    done
-fi
-
-# part added by Stephen Mrenna to correct the kinematics of the replaced
-#  particles
-if [[ -e ./madevent/bin/internal/addmasses.py ]]; then
-  mv ./events.lhe ./events.lhe.0
-  python ./madevent/bin/internal/addmasses.py ./events.lhe.0 ./events.lhe
-fi  
-
 gzip -f events.lhe
 exit
=== modified file 'MG5_aMC_v2_2_1/MadSpin/decay.py'
--- MadSpin/decay.py    2014-09-17 09:53:23 +0000
+++ MG5_aMC_v2_2_1/MadSpin/decay.py    2014-10-14 11:29:10 +0000
@@ -1185,7 +1185,7 @@
             decaying_base = [id for id in final if abs(id) in self.decay_ids]
             decaying_base.sort()
             topo['base_order'] = (initial , final)
-            topo['matrix_element'] = me
+#            topo['matrix_element'] = me
             tags = []
             topo['tag2order'] = {}
              
@@ -2791,11 +2791,11 @@
                                          'processes': matrix_element.get('processes'),
                                          'tag': me.shell_string(pdg_order=True)}
         #
-        if __debug__:
-            #check that all decay matrix element correspond to a decay only
-            for prod in self.all_ME.values():
-                for decay in prod['matrix_element']['base_amplitude']['process']['decay_chains']:
-                    assert decay.shell_string() in self.all_decay
+#        if __debug__:
+#            #check that all decay matrix element correspond to a decay only
+#            for prod in self.all_ME.values():
+#                for decay in prod['matrix_element']['base_amplitude']['process']['decay_chains']:
+#                    assert decay.shell_string() in self.all_decay
             
         
     def get_branching_ratio(self):

=== modified file 'MG5_aMC_v2_2_1/MadSpin/decay.py'
--- MadSpin/decay.py    2014-09-17 09:53:23 +0000
+++ MG5_aMC_v2_2_1/MadSpin/decay.py    2014-10-16 13:08:38 +0000
@@ -3221,14 +3221,23 @@
             helicities=[lastline[i] for i in range(len(lastline))]
             output = trials, BWvalue, weight, momenta, failed, use_mc_masses, helicities
 
-        if len(self.calculator) > 100:
-            logger.debug('more than 100 calculator. Perform cleaning')
+        if len(self.calculator) > self.options['max_running_process']:
+            logger.debug('more than %s calculators. Perform cleaning' % self.options['max_running_process'])
             nb_calls = self.calculator_nbcall.values()
             nb_calls.sort()
             cut = max([nb_calls[len(nb_calls)//2], 0.001 * nb_calls[-1]])
             for key, external in list(self.calculator.items()):
                 nb = self.calculator_nbcall[key]
                 if nb < cut:
+                    if key[0]=='full':
+                      path=key[1]
+                      end_signal="5 0 0 0 \n"  # before closing, write down the seed 
+                      external.stdin.write(end_signal)
+                      ranmar_state=external.stdout.readline()
+                      ranmar_file=pjoin(path,'ranmar_state.dat')
+                      ranmar=open(ranmar_file, 'w')
+                      ranmar.write(ranmar_state)
+                      ranmar.close()
                     external.stdin.close()
                     external.stdout.close()
                     external.terminate()


=== modified file 'MG5_aMC_v2_2_1/madgraph/interface/madevent_interface.py'
--- madgraph/interface/madevent_interface.py    2014-09-24 19:41:18 +0000
+++ MG5_aMC_v2_2_1/madgraph/interface/madevent_interface.py    2014-10-17 15:57:21 +0000
@@ -4064,11 +4064,10 @@
             
 
             logfile = pjoin(Pdir, 'gen_ximprove.log')
-            proc = misc.Popen([pjoin(bindir, 'gen_ximprove')],
+            misc.call([pjoin(bindir, 'gen_ximprove')],
                                     stdin=subprocess.PIPE,
                                     stdout=open(logfile,'w'),
                                     cwd=Pdir)
-            proc.communicate('%s 1 F\n' % (precision))
 
             if os.path.exists(pjoin(Pdir, 'ajob1')):
                 alljobs = glob.glob(pjoin(Pdir,'ajob*'))
