--- MG5_aMC_v2_3_2_2/madgraph/interface/reweight_interface.py	2015-09-07 20:07:39.000000001 +0200
+++ MG5_aMC_v2_3_2_2_patch/madgraph/interface/reweight_interface.py	2015-10-12 16:20:14.000000001 +0200
@@ -734,7 +734,8 @@
                         dir_to_f2py_free_mod[(Pdir,0)] = (metag, nb_f2py_module)
                 
                 os.environ['MENUM'] = '2'
-                misc.compile(['matrix2py.so'], cwd=Pdir)
+                if not self.rwgt_dir or not os.path.exists(pjoin(Pdir, 'matrix2py.so')):
+                    misc.compile(['matrix2py.so'], cwd=Pdir)
                 
                 self.rename_f2py_lib(Pdir, 2*metag)
                 with misc.chdir(Pdir):
@@ -755,7 +756,8 @@
                     os.remove(pjoin(Pdir, 'matrix%spy.so' % newtag))
                     newtag  = "L%s" % newtag
                     os.environ['MENUM'] = newtag
-                    misc.compile(['matrix%spy.so' % newtag], cwd=Pdir)
+                    if not self.rwgt_dir or not os.path.exists(pjoin(Pdir,'matrix%spy.so' % newtag )):
+                        misc.compile(['matrix%spy.so' % newtag], cwd=Pdir)
                     mymod = __import__('rw_me.SubProcesses.%s.matrix%spy' % (Pname, newtag), globals(), locals(), [],-1)
                 
                 S = mymod.SubProcesses
@@ -773,7 +775,8 @@
             assert hypp_id == 1
             Pname = os.path.basename(Pdir)
             os.environ['MENUM'] = '2'
-            misc.compile(['matrix2py.so'], cwd=pjoin(subdir, Pdir))
+            if not self.rwgt_dir or not os.path.exists(pjoin(Pdir, 'matrix2py.so')):
+                misc.compile(['matrix2py.so'], cwd=pjoin(subdir, Pdir))
             if (Pdir, 1) not in dir_to_f2py_free_mod:
                 metag = 1
                 dir_to_f2py_free_mod[(Pdir,1)] = (metag, nb_f2py_module)
@@ -790,7 +793,8 @@
                     os.remove(pjoin(Pdir, 'matrix%spy.so' % metag ))
                     metag = "L%s" % metag
                     os.environ['MENUM'] = str(metag)
-                    misc.compile(['matrix%spy.so' % metag], cwd=pjoin(subdir, Pdir))
+                    if not self.rwgt_dir or not os.path.exists(pjoin(Pdir,'matrix%spy.so' % metag)):
+                        misc.compile(['matrix%spy.so' % metag], cwd=pjoin(subdir, Pdir))
                     mymod = __import__("rw_me_second.SubProcesses.%s.matrix%spy" % (Pname, metag))
                     
                 reload(mymod)
