--- ../old/MG5_aMC_v2_3_2_2/madgraph/interface/reweight_interface.py	2015-09-07 20:07:39.000000001 +0200
+++ MG5_aMC_v2_3_2_2/madgraph/interface/reweight_interface.py	2015-09-18 18:23:49.000000001 +0200
@@ -737,8 +737,8 @@
                 misc.compile(['matrix2py.so'], cwd=Pdir)
                 
                 self.rename_f2py_lib(Pdir, 2*metag)
+                mymod = __import__('rw_me.SubProcesses.%s.matrix%spy' % (Pname, 2*metag), globals(), locals(), [],-1)
                 with misc.chdir(Pdir):
-                    mymod = __import__('rw_me.SubProcesses.%s.matrix%spy' % (Pname, 2*metag), globals(), locals(), [],-1)
                     S = mymod.SubProcesses
                     P = getattr(S, Pname)
                     mymod = getattr(P, 'matrix%spy' % (2*metag))
