--- a/madgraph/various/systematics.py	2017-08-16 17:25:24.000000000 -0500
+++ b/madgraph/various/systematics.py	2018-02-20 19:33:49.455432428 -0600
@@ -479,7 +479,7 @@
             else:
                 resume.write( '#PDF %s: %g +%2.3g%% -%2.3g%%\n' % (pdfset.name, pdferr.central,pdferr.errplus*100/all_cross[0], pdferr.errminus*100/all_cross[0]))
 
-        dyn_name = {1: '\sum ET', 2:'\sum\sqrt{m^2+pt^2}', 3:'0.5 \sum\sqrt{m^2+pt^2}',4:'\sqrt{\hat s}' }
+        dyn_name = {1: '\sum ET', 2:'\sum\sqrt{m^2+pt^2}', 3:'0.5 \sum\sqrt{m^2+pt^2}',4:'\sqrt{\hat s}', 5:'max{pt(j1), pt(j2)}' }
         for key, curr in dyns.items():
             if key ==-1:
                 continue
@@ -582,7 +582,7 @@
                 info += 'alpsfact=%s ' % alps
             if dyn!=-1.:
                 tag += 'DYN_SCALE="%s" ' % dyn
-                info += 'dyn_scale_choice=%s ' % {1:'sum pt', 2:'HT',3:'HT/2',4:'sqrts'}[dyn]
+                info += 'dyn_scale_choice=%s ' % {1:'sum pt', 2:'HT',3:'HT/2',4:'sqrts',5:'max ptj'}[dyn]
                                            
             if pdf != self.orig_pdf:
                 tag += 'PDF="%s" ' % pdf.lhapdfID
@@ -750,6 +750,8 @@
                 mur = event.get_ht_scale(0.5)
             elif dyn == 4:
                 mur = event.get_sqrts_scale(1.)
+            elif dyn == 5:
+                mur = event.get_maxptj_scale(1.)
             muf1 = mur
             muf2 = mur
             loinfo = dict(loinfo)
