--- init_couplings.f.org	2015-07-30 14:35:19.000000001 +0200
+++ init_couplings.f	2015-09-18 17:05:38.000000001 +0200
@@ -88,12 +88,14 @@
 
 c     set mass windows around the resonance peak 
 c     It is used in the generation of the Born phase space
       masswindow = 30
+      masswindow = powheginput("#masswindow")
+      if(masswindow.lt.0d0) masswindow=30d0
 c      ph_Zmass2low=(ph_Zmass-masswindow*ph_Zwidth)**2
 c      ph_Zmass2high=(ph_Zmass+masswindow*ph_Zwidth)**2
       ph_Hmass2low=max(0d0,ph_Hmass-masswindow*ph_Hwidth)
       ph_Hmass2low=ph_Hmass2low**2
-      ph_Hmass2high=(ph_Hmass+masswindow*ph_Hwidth)**2
+      ph_Hmass2high=min(kn_sbeams,(ph_Hmass+masswindow*ph_Hwidth)**2)
 c      ph_Hmass2low=0d0
 c      ph_Hmass2high=kn_sbeams/4
     
