Index: PhysPars.h
===================================================================
--- PhysPars.h	(revision 3720)
+++ PhysPars.h	(working copy)
@@ -1,12 +1,12 @@
 c -*- Fortran -*-
-      real * 8 ph_alphaem,ph_Zmass,ph_Zwidth,ph_Wmass,ph_Wwidth,ph_cthw,
+      real * 8 ph_alphaem,ph_Zmass,ph_Zwidth,osZwidth,ph_Wmass,ph_Wwidth,osWwidth,ph_cthw,
      $     ph_sthw,ph_sthw2,ph_Zmass2,ph_Zmass2low,ph_Zmass2high,
      $     ph_Wmass2,ph_Wmass2low,ph_Wmass2high ,ph_ZmZw,ph_WmWw,
-     $     ph_unit_e, ph_CKM(3,3)
-      common/ph_common/ph_alphaem,ph_Zmass,ph_Zwidth,ph_Wmass,ph_Wwidth,
+     $     ph_unit_e, ph_gmu, ph_CKM(3,3)
+      common/ph_common/ph_alphaem,ph_Zmass,ph_Zwidth,osZwidth,ph_Wmass,ph_Wwidth,osWwidth,
      $     ph_cthw,ph_sthw,ph_sthw2,ph_Zmass2,ph_Zmass2low,
      $     ph_Zmass2high,ph_Wmass2,ph_Wmass2low,ph_Wmass2high ,ph_ZmZw,
-     $     ph_WmWw,ph_unit_e,ph_CKM
+     $     ph_WmWw,ph_unit_e,ph_gmu,ph_CKM
 
