diff -ruN ./makefile ../superchic_v42_patch/makefile
--- ./makefile	2023-03-08 10:57:13.000000001 +0100
+++ ../superchic_v42_patch/makefile	2023-12-06 23:36:57.000000001 +0100
@@ -1,7 +1,7 @@
 LIBFLAGlha = -lLHAPDF
 LIBFLAGapfel = -lAPFEL -lAPFELevol
 LHAPDFLIB = `lhapdf-config --prefix`/lib
-APFELLIB = `apfel-config --prefix`/lib
+APFELLIB = ${APFEL_LIBRARY_PATH}
 FC = gfortran -g -w -Wall -fbacktrace -fcheck=all -fPIC -ffixed-line-length-132
 
 #####################
