--- a/madgraph/various/systematics.py	2019-05-29 10:48:35.000000001 +0200
+++ b/madgraph/various/systematics.py	2019-05-28 16:02:38.000000001 +0200
@@ -314,7 +314,7 @@
         else:
             lowest_id = self.get_id()        
 
-        ids = [lowest_id+i for i in range(len(self.args)-1)]
+        ids = [lowest_id+i for i in range(len(self.args))]
         all_cross = [0 for i in range(len(self.args))]
         
         self.input.parsing = False
@@ -348,7 +348,7 @@
                 print event
                 raise Exception
             
-            wgt = [event.wgt*wgts[i]/wgts[0] for i in range(1,len(wgts))]
+            wgt = [event.wgt*wgts[i]/wgts[0] for i in range(0,len(wgts))]
             all_cross = [(all_cross[j] + event.wgt*wgts[j]/wgts[0]) for j in range(len(wgts))]
             
             rwgt_data = event.parse_reweight()
@@ -512,9 +512,9 @@
         text = ''
         
         default = self.args[0]
-        for arg in self.args[1:]:
+        for arg in self.args:
             mur, muf, alps, dyn, pdf = arg[:5]
-            if pdf == self.orig_pdf and alps ==1 and (mur!=1 or muf!=1 or dyn!=-1):
+            if (pdf == self.orig_pdf and alps ==1 and (mur!=1 or muf!=1 or dyn!=-1)) or (cid == lowest_id):
                 if not in_scale:
                     text += "<weightgroup name=\"Central scale variation\" combine=\"envelope\">\n"
                     in_scale=True
@@ -533,7 +533,7 @@
                 text += "</weightgroup> # ALPS\n"
                 in_alps=False
             
-            if mur == muf == 1 and dyn==-1 and alps ==1:
+            if mur == muf == 1 and dyn==-1 and alps ==1 and (cid != lowest_id):
                 if pdf.lhapdfID < 0:
                     for central,sets in self.pdfsets.items():
                         if pdf in sets.set():
