diff --git a/MadSpin/interface_madspin.py b/MadSpin/interface_madspin.py
index d183576..134ebda 100755
--- a/MadSpin/interface_madspin.py
+++ b/MadSpin/interface_madspin.py
@@ -717,6 +717,7 @@ class MadSpinInterface(extended_cmd.Cmd):
         # NOW we have all the information available for RUNNING
         
         if self.options['seed']:
+            self.options['seed'] = self.options['seed']
             #seed is specified need to use that one:
             open(pjoin(self.options['ms_dir'],'seeds.dat'),'w').write('%s\n'%self.options['seed'])
             #remove all ranmar_state
